File: frontend/src/types/documentation.ts
export interface OutreachDocumentation {
  id: number

  hospital_id: number
  patient_id: number
  call_id: number

  triage_assessment_id: number | null
  consensus_assessment_id: number | null
  escalation_id: number | null

  summary: string

  symptoms: string
  medication_concerns: string
  patient_questions: string
  red_flags: string

  triage_risk_level: string
  recommended_action: string

  escalation_created: boolean
  escalation_reason: string | null

  follow_up_required: boolean
  follow_up_notes: string | null

  created_at: string
}

File: ai/schemas/documentation.py
from pydantic import BaseModel, Field


class OutreachDocumentation(BaseModel):
    summary: str

    symptoms: list[str] = Field(
        default_factory=list
    )

    medication_concerns: list[str] = Field(
        default_factory=list
    )

    patient_questions: list[str] = Field(
        default_factory=list
    )

    red_flags: list[str] = Field(
        default_factory=list
    )

    triage_risk_level: str

    recommended_action: str

    escalation_created: bool

    escalation_reason: str | None = None

    follow_up_required: bool = False

    follow_up_notes: str | None = None
